gitlab/pkg/adapter/receive_adapter.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adapter
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/webhooks.v5/gitlab"
+	"knative.dev/eventing/pkg/adapter"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/source"
+)
+
+const (
+	glHeaderToken = "X-Gitlab-Token"
+	glHeaderEvent = "X-Gitlab-Event"
+)
+
+type envConfig struct {
+	adapter.EnvConfig
+
+	// Environment variable containing Gitlab secret token
+	EnvSecret string `envconfig:"GITLAB_SECRET_TOKEN" required:"true"`
+	// Port to listen incoming connections
+	Port string `envconfig:"PORT" default:"8080"`
+}
+
+// gitLabReceiveAdapter converts incoming GitLab webhook events to
+// CloudEvents and then sends them to the specified Sink
+type gitLabReceiveAdapter struct {
+	logger      *zap.SugaredLogger
+	client      cloudevents.Client
+	secretToken string
+	port        string
+}
+
+// NewEnvConfig function reads and parses
+func NewEnvConfig() adapter.EnvConfigAccessor {
+	return &envConfig{}
+}
+
+// NewAdapter rturns the instance of gitLabReceiveAdapter that implements Start() method
```

Comment:
Typo "returns"

(nit) It doesn't implement a method but the adapter.Adapter interface.